library/core/lifecycle-events/src/main/java/org/quiltmc/qsl/lifecycle/api/client/event/ClientLifecycleEvents.java

```diff
@@ -0,0 +1,97 @@
+package org.quiltmc.qsl.lifecycle.api.client.event;
+
+import org.quiltmc.qsl.base.api.event.ArrayEvent;
+
+import net.minecraft.client.MinecraftClient;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Events indicating the lifecycle of a Minecraft client.
+ *
+ * <p>The lifecycle of a Minecraft client, starts when the client is ready. The client will tick the client and then the
+ * integrated server if in single player.
+ *
+ * @see ClientTickEvents
+ */
+@Environment(EnvType.CLIENT)
+public final class ClientLifecycleEvents {
+	// There is no CLIENT_STARTING event because there is no way to allow mods to register callbacks that early without
+	// possibly initializing the game's registries improperly in preLaunch.
+
+	/**
+	 * An event indicating that a Minecraft client is ready to tick and render.
+	 *
+	 * <p>It should be noted this event is executed while the splash screen is visible, not the main menu.
+	 */
+	public static final ArrayEvent<Ready> READY = ArrayEvent.create(Ready.class, callbacks -> client -> {
+		for (var callback : callbacks) {
+			callback.readyClient(client);
+		}
+	});
+
+	/**
+	 * An event indicating that a Minecraft client has finished it's last tick and will shut down.
+	 *
+	 * <p>If the client is connected to a server, the client will disconnect from the server. Then if the client has a
+	 * running an integrated server, the integrated server will be shutdown. Finally all client facilities are torn down.
+	 *
+	 * <h2>What should mods do when this event is executed</h2>
+	 *
+	 * Mods which maintain session data when connected to a server should should save that data here as those mods still
+	 * have access to the connected server.
+	 *
+	 * <p>If your mod has any data on the integrated server, avoid doing that here, use
+	 * {@link org.quiltmc.qsl.lifecycle.api.event.ServerLifecycleEvents#STOPPING ServerLifecycleEvents.SERVER_STOPPING}
```

Comment:
Why are you renaming `STOPPING` to `SERVER_STOPPING` here?
```suggestion
	 * {@link org.quiltmc.qsl.lifecycle.api.event.ServerLifecycleEvents#STOPPING ServerLifecycleEvents.STOPPING}
```